hw1-app/src/app/courses/course-item/course-item.component.ts

```diff
@@ -1,25 +1,74 @@
-import { Component, OnInit, ChangeDetectionStrategy } from '@angular/core';
-import { ICourse, Course } from '../models/course-item.model';
+import { Component,
+  OnInit,
+  ChangeDetectionStrategy,
+  Input,
+  EventEmitter,
+  Output,
+  OnChanges,
+  DoCheck,
+  AfterContentInit,
+  AfterContentChecked,
+  AfterViewInit,
+  AfterViewChecked,
+  OnDestroy } from '@angular/core';
+import { ICourse } from '../models/course-item.model';
 
 @Component({
   selector: 'course-item',
   templateUrl: './course-item.component.html',
   styleUrls: ['./course-item.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class CourseItemComponent implements OnInit {
-  public item: ICourse = new Course({
-    id: 'id',
-    title: 'title',
-    creationDate: new Date(),
-    duration: 33,
-    description: 'description'
-  });
+export class CourseItemComponent implements OnInit,
+    OnChanges,
+    DoCheck,
+    AfterContentInit,
+    AfterContentChecked,
+    AfterViewInit,
+    AfterViewChecked,
+    OnDestroy {
+  @Input() item: any = {};
+  @Output() deleteCourse: EventEmitter<ICourse> = new EventEmitter();
```

Comment:
You can write shorter 
 @Output() deleteCourse = new EventEmitter<ICourse>();

Reply:
done